File: Python/Harddrive.py
from Komponen import Komponen
from ElektronikDevice import ElektronikDevice

class Harddrive(Komponen, ElektronikDevice):
    __kapasitas_gb = 0
    __tipe_drive = ""
    __kecepatan_mbps = 0

    def __init__(self, kapasitas_gb=0, tipe_drive="", kecepatan_mbps=0, merk="", nama="", konsumsi_daya_watt=0):
        Komponen.__init__(self, merk, nama)
        ElektronikDevice.__init__(self, konsumsi_daya_watt)  # Fixed bug here
        self.__kapasitas_gb = kapasitas_gb
        self.__tipe_drive = tipe_drive
        self.__kecepatan_mbps = kecepatan_mbps

    def set_kapasitas_gb(self, kapasitas_gb):
        self.__kapasitas_gb = kapasitas_gb

    def set_tipe_drive(self, tipe_drive):
        self.__tipe_drive = tipe_drive

    def set_kecepatan_mbps(self, kecepatan_mbps):
        self.__kecepatan_mbps = kecepatan_mbps

    def get_kapasitas_gb(self):
        return self.__kapasitas_gb
    
    def get_tipe_drive(self):
        return self.__tipe_drive
    
    def get_kecepatan_mbps(self):
        return self.__kecepatan_mbps

File: Java/Harddrive.java
public class Harddrive extends Komponen {

    private int kapasitasGB;
    private String tipeDrive;
    private int kecepatanMBps;
    private ElektronikDevice elektronikDevice;

    public Harddrive() {
        super();
        this.elektronikDevice = new ElektronikDevice();
        this.kapasitasGB = 0;
        this.tipeDrive = "";
        this.kecepatanMBps = 0;
    }

    public Harddrive(int kapasitasGB, String tipeDrive, int kecepatanMBps, String merk, String nama, int konsumsiDayaWatt) {
        super(merk, nama);
        this.elektronikDevice = new ElektronikDevice(konsumsiDayaWatt);
        this.kapasitasGB = kapasitasGB;
        this.tipeDrive = tipeDrive;
        this.kecepatanMBps = kecepatanMBps;
    }

    public void setKapasitasGB(int kapasitasGB) {
        this.kapasitasGB = kapasitasGB;
    }

    public void setTipeDrive(String tipeDrive) {
        this.tipeDrive = tipeDrive;
    }

    public void setKecepatanMBps(int kecepatanMBps) {
        this.kecepatanMBps = kecepatanMBps;
    }
    
    public void setKonsumsiDayaWatt(int konsumsiDayaWatt) {
        this.elektronikDevice.setKonsumsiDayaWatt(konsumsiDayaWatt);
    }

    public int getKapasitasGB() {
        return kapasitasGB;
    }

    public String getTipeDrive() {
        return tipeDrive;
    }

    public int getKecepatanMBps() {
        return kecepatanMBps;
    }
    
    public int getKonsumsiDayaWatt() {
        return this.elektronikDevice.getKonsumsiDayaWatt();
    }
}